### PR TITLE
stop gradients

### PR DESCRIPTION
**Pull request by EderSantana**
stop gradient operations are used for complicated graphs such as RNNs that feed previous outputs as new inputs:
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/seq2seq.py#L103

### CODE REVIEW

**Comment by EderSantana** [comment]
for the time being, this PR mostly works for the purposes of keras backend as an unifying UI for theano and tensorflow. But it should be useful later for building such models without having to compile one step and feeding outputs and hidden states manually for the RNN

**Comment by fchollet** [comment]
Sounds good to me. Somehow the tests appear to be failing, though. Hard to make sense of that error.

**Comment by EderSantana** [comment]
I think I found the error @fchollet. I'll keep iterating here with Travis because if I use my gpus while the tests are running their results get all messed up. I'll call you if I need any help or when the PR is ready. Sounds good?

**Comment by EderSantana** [comment]
@farizrahman4u I think this might be relevant for your `seq2seq` project. I believe the LSTM decoder should have `stop_gradient` when consuming its own input. What has been your experience with this subject?

**Comment by farizrahman4u** [comment]
Sounds great!

**Comment by EderSantana** [comment]
@fchollet  seems like everything is fine here!

**Comment by fchollet** [comment]
LGTM
